full-node/sov-ethereum/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+use borsh::ser::BorshSerialize;
+use const_rollup_config::ROLLUP_NAMESPACE_RAW;
+use demo_stf::app::DefaultPrivateKey;
+use demo_stf::runtime::{DefaultContext, Runtime};
+use ethers::types::transaction::eip2718::TypedTransaction;
+use ethers::types::Bytes;
+use ethers::utils::rlp::Rlp;
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::core::params::ArrayParams;
+use jsonrpsee::http_client::{HeaderMap, HttpClient};
+use jsonrpsee::RpcModule;
+use jupiter::da_service::DaServiceConfig;
+use sov_evm::call::CallMessage;
+use sov_evm::evm::EvmTransaction;
+use sov_modules_api::transaction::Transaction;
+
+const GAS_PER_BYTE: usize = 120;
+
+pub fn get_ethereum_rpc(config: DaServiceConfig) -> RpcModule<Ethereum> {
+    let mut rpc = RpcModule::new(Ethereum { config });
+    register_rpc_methods(&mut rpc).expect("Failed to register sequencer RPC methods");
+    rpc
+}
+
+pub struct Ethereum {
+    config: DaServiceConfig,
+}
+
+impl Ethereum {
+    fn make_client(&self) -> HttpClient {
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            "Authorization",
+            format!("Bearer {}", self.config.celestia_rpc_auth_token.clone())
+                .parse()
+                .unwrap(),
+        );
+
+        jsonrpsee::http_client::HttpClientBuilder::default()
+            .set_headers(headers)
+            .max_request_body_size(default_max_response_size()) // 100 MB
+            .build(self.config.celestia_rpc_address.clone())
+            .expect("Client initialization is valid")
+    }
+
+    async fn send_tx_to_da(
```

Comment:
Just an idea, but since Sequencer is merged, it can be used to create blobs.

Reply:
Also, this functionality is available in DaLaayer service. So current implementation depends on Celestia, where if it uses trait, it should work on all Da Layers.

Reply:
Yes I will extract the common functionality in the next PR.